src/tptp_parser.rs

```diff
@@ -181,7 +545,148 @@ pub fn parse_file<'a>(file: PathBuf) -> TPTPProblem {
     TPTPProblem {
         axioms: axioms,
         conjectures: conjectures,
-        negated_conjectures: negated_conjectures,
+    }
+}
+
+// Transform the TPTPProblem into the problem in CNF for our saturation prover:
+// - we conjunct the assumption formulas
+// - we conject those with the negated goals
+// - we transform it into CNF
+// - we show False
+pub fn transform_problem(problem: TPTPProblem) -> SkolemTerm {
+    let mut acc;
+    if problem.conjectures.len() != 0 {
+        let neg_goals: Vec<FOLTerm> = problem
+            .conjectures
+            .into_iter()
+            .map(FOLTerm::negate)
+            .collect();
+        log::info!("Neg. Goals: {:?}", neg_goals);
+        acc = FOLTerm::from(neg_goals[0].clone());
+        for t in neg_goals[1..].into_iter() {
+            acc = FOLTerm::And(Box::new(acc), Box::new(t.clone()));
+        }
+        for t in problem.axioms {
+            acc = FOLTerm::And(Box::new(acc), Box::new(t.clone()));
+        }
+    } else {
+        log::warn!("The TPTP Problem contains no goals!");
+        acc = FOLTerm::from(problem.axioms[0].clone());
+        for t in problem.axioms[1..].into_iter() {
+            acc = FOLTerm::And(Box::new(acc), Box::new(t.clone()));
+        }
+    }
+    let skolem_term = SkolemTerm::from(acc);
+    log::info!("SkolemTerm: {}", skolem_term);
+    let cnf_term = cnf(skolem_term.clone());
+    log::info!("CNF Term: {}", skolem_term);
+    cnf_term
+}
+
+fn add_term_to_termbank(
```

Comment:
It might be nice to pull this stuff out into a struct instead that maintain the term bank and the two maps as member fields and has the if let Some(var) = ... code as some cute separate function so the intention of the transformation is more clear.